### PR TITLE
ref(tests): update rule processor test to use GroupEvent

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Fairly innocuous change, the actual code already makes use of `GroupEvent` in event_manager, this just updates the tests to reflect that.

### CODE REVIEW

File: tests/sentry/rules/test_processor.py [hunk 1]
```diff
@@ -246,11 +246,13 @@ class RuleProcessorTestFilters(TestCase):
         "tests.sentry.rules.test_processor.MockFilterFalse",
     )
 
+    def setUp(self):
+        self.event = self.store_event(data={}, project_id=self.project.id)
+        self.event = next(self.event.build_group_events())
```
Comment (rjo100) [inline tests/sentry/rules/test_processor.py:251]: worth changing this variable name to reflect that it's a group event? might be a separate PR due to size of change though with line 53 as well

**Review by rjo100 — APPROVED** [review]

File: tests/sentry/rules/test_processor.py (hunk 1 above)
Comment (barkbarkimashark) [reply tests/sentry/rules/test_processor.py]: Yup, good idea. Just added.
